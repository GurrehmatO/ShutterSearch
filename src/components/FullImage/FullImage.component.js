import React, { Component } from 'react';
import { ActivityIndicator, Dimensions } from 'react-native';
import { Card, Text, Image } from 'react-native-elements';

class FullImage extends Component {
  render() {
    const { navigation } = this.props;
    const item = navigation.getParam('item', {});
    return (
      <Card>
        <Image
          source={{ uri: item.assets.preview_1500.url }}
          PlaceholderContent={(<ActivityIndicator />)}
          style={{
            width: '100%',
            height: Dimensions.get('window').width * 0.87 / item.aspect,
          }}
        />
        <Text>{item.description}</Text>
      </Card>
    );
  }
}

export default FullImage;
