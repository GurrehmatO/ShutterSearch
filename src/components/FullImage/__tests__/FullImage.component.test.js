import React from 'react';
import renderer from 'react-test-renderer';
import FullImage from '../FullImage.component';

describe('FullImage component', () => {
  it('should render correctly', () => {
    const props = {
      navigation: {
        getParam: () => (
          {
            id: '1052251895',
            aspect: 1.5,
            assets: {
              preview: {
                height: 300,
                url: 'https://image.shutterstock.com/display_pic_with_logo/154447/1052251895/stock-photo-child-in-pajamas-relaxing-on-a-window-sill-with-pet-lazy-weekend-with-cat-at-home-cozy-scene-1052251895.jpg',
                width: 450,
              },
              small_thumb: {
                height: 67,
                url: 'https://thumb9.shutterstock.com/thumb_small/154447/1052251895/stock-photo-child-in-pajamas-relaxing-on-a-window-sill-with-pet-lazy-weekend-with-cat-at-home-cozy-scene-1052251895.jpg',
                width: 100,
              },
              large_thumb: {
                height: 100,
                url: 'https://thumb9.shutterstock.com/thumb_large/154447/1052251895/stock-photo-child-in-pajamas-relaxing-on-a-window-sill-with-pet-lazy-weekend-with-cat-at-home-cozy-scene-1052251895.jpg',
                width: 150,
              },
              huge_thumb: {
                height: 260,
                url: 'https://image.shutterstock.com/image-photo/child-pajamas-relaxing-on-window-260nw-1052251895.jpg',
                width: 390,
              },
              preview_1000: {
                url: 'https://ak.picdn.net/shutterstock/photos/1052251895/watermark_1000/faec05ee0949077d9f033e3ee91883e6/preview_1000-1052251895.jpg',
                width: 1000,
                height: 667,
              },
              preview_1500: {
                url: 'https://image.shutterstock.com/z/stock-photo-child-in-pajamas-relaxing-on-a-window-sill-with-pet-lazy-weekend-with-cat-at-home-cozy-scene-1052251895.jpg',
                width: 1500,
                height: 1000,
              },
            },
            contributor: {
              id: '154447',
            },
            description: 'Child in pajamas relaxing on a window sill with pet. Lazy weekend with cat at home. Cozy scene, hygge concept.',
            image_type: 'photo',
            has_model_release: false,
            media_type: 'image',
          }),
      },
    };
    const component = renderer.create(<FullImage {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
