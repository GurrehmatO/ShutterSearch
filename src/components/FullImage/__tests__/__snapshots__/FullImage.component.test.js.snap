// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullImage component should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "position": "relative",
        }
      }
    >
      <Image
        onLoadEnd={[Function]}
        source={
          Object {
            "uri": "https://image.shutterstock.com/z/stock-photo-child-in-pajamas-relaxing-on-a-window-sill-with-pet-lazy-weekend-with-cat-at-home-cozy-scene-1052251895.jpg",
          }
        }
        style={
          Object {
            "height": 435,
            "width": "100%",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      />
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "height": 435,
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="RNE__Image__placeholder"
        >
          <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="small"
          />
        </View>
      </View>
    </View>
    <Text
      style={Object {}}
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Child in pajamas relaxing on a window sill with pet. Lazy weekend with cat at home. Cozy scene, hygge concept.
    </Text>
  </View>
</View>
`;
